internal/rpcserver/rpcserver.go

```diff
@@ -3016,6 +3064,31 @@ func handleGetTicketPoolValue(_ context.Context, s *Server, cmd interface{}) (in
 	return amt.ToCoin(), nil
 }
 
+// handleGetTreasuryBalance implements the gettreasurybalance command.
+func handleGetTreasuryBalance(_ context.Context, s *Server, cmd interface{}) (interface{}, error) {
```

Comment:
I'm fixing this as part of addressing all the issues with the `TreasuryBalance` func in `blockchain`, but this function is full of incorrect results:

* Wrong error code and useless error message for malformed hashes
```
$ ./ctl gettreasurybalance foo
-32603: encoding/hex: invalid byte: U+006F 'o'
```

* Wrong error code on unknown blocks
```
$ ./ctl gettreasurybalance 1
-32603: block 1 is not known
```

* Wrong error code on inactive treasury for block 0
```
$ ./ctl getblockhash 0 | ./ctl gettreasurybalance -
-32603: treasury not active
```

* Off by one in the detection of treasury agenda state leading to incorrect error
```
$ ./ctl getblockhash 1 | ./ctl gettreasurybalance -
-32603: treasury db missing key: 01f34c776d4fd8290eab87247735e741f1ad3d5b179ea81a8128ce0f165ef7c2
```

Reply:
With my updates:

```
$ ./ctl gettreasurybalance
{
  "hash": "0271dd1f49ba977754671fbb75543c92089c28e06f869fbd003c4613ec5054cc",
  "height": 32,
  "balance": 75000000000
}

$ ./ctl gettreasurybalance foo
-22: Argument must be hexadecimal string (not "foo")

$ ./ctl gettreasurybalance 1
-5: Block not found: 0000000000000000000000000000000000000000000000000000000000000001

$ ./ctl getblockhash 0 | ./ctl gettreasurybalance -
-5: Treasury inactive for block 6bef82c645999585f7255cb02672921ac2f5492820090cd635fe3a59d16b4f87

$ ./ctl getblockhash 1 | ./ctl gettreasurybalance -
-5: Treasury inactive for block 41c5ada13e81f2eaa9f97786537b8e2ff0e487f5e279c89856c5348db5e280c4

$ ./ctl getblockhash 2 | ./ctl gettreasurybalance -
{
  "hash": "784e60e84ff311ade5bda635a26402eb9645f43719f626a93434184d1cc95777",
  "height": 2,
  "balance": 0
}
```